fuse_optimizers/include/fuse_optimizers/fixed_lag_smoother_params.h

```diff
@@ -0,0 +1,116 @@
+/***************************************************************************
+ * Copyright (C) 2019 Locus Robotics. All rights reserved.
+ * Unauthorized copying of this file, via any medium, is strictly prohibited
+ * Proprietary and confidential
+ ***************************************************************************/
+#ifndef FUSE_OPTIMIZERS_FIXED_LAG_SMOOTHER_PARAMS_H
+#define FUSE_OPTIMIZERS_FIXED_LAG_SMOOTHER_PARAMS_H
+
+#include <ros/console.h>
+#include <ros/duration.h>
+#include <ros/node_handle.h>
+
+#include <algorithm>
+#include <string>
+#include <vector>
+
+
+namespace fuse_optimizers
+{
+
+/**
+ * @brief Defines the set of parameters required by the fuse_optimizers::FixedLagSmoother class
+ */
+struct FixedLagSmootherParams
+{
+public:
+  /**
+   * @brief The set of sensors whose transactions will trigger the optimizer thread to start running
+   *
+   * This is designed to keep the system idle until the origin constraint has been received.
+   */
+  std::vector<std::string> ignition_sensors;
+
+  /**
+   * @brief The duration of the smoothing window in seconds
+   */
+  ros::Duration lag_duration { 5.0 };
+
+  /**
+   * @brief The target duration for optimization cycles
+   *
+   * If an optimization takes longer than expected, an optimization cycle may be skipped. The optimization period
+   * may be specified in either the "optimization_period" parameter in seconds, or in the "optimization_frequency"
+   * parameter in Hz.
+   */
+  ros::Duration optimization_period { 0.1 };
+
+  /**
+   * @brief The topic name of the advertised reset service
+   */
+  std::string reset_service { "reset" };
+
+  /**
+   * @brief The maximum time to wait for motion models to be generated for a received transactions.
+   *
+   * Transactions are processes sequentially, so no new transactions will be added to the graph while waiting for
+   * motion models to be generated. Once the timeout expires, that transaction will be deleted from the queue.
+   */
+  ros::Duration transaction_timeout { 0.1 };
+
+  /**
+   * @brief Helper function that loads strictly positive floating point values from the parameter server
+   *
+   * @param[in] node_handle - The node handle used to load the parameter
+   * @param[in] parameter_name - The parameter name to load
+   * @param[in] default_value - A default value to use if the provided parameter name does not exist
+   * @return The loaded (or default) value
+   */
+  double getPositiveParam(const ros::NodeHandle& node_handle, const std::string& parameter_name, double default_value)
```

Comment:
Not for this PR, but if we can't release some of our internal utility methods like this, we should consider adding some of them to a utility header in this package.